src/dataregistry/query.py

```diff
@@ -0,0 +1,248 @@
+
+        # Get table definitions
+        self._tables = dict()
+        self._tables["dataset"] = self._metadata_getter.get("dataset")
+        self._tables["execution"] = self._metadata_getter.get("execution")
+        # self._dataset_alias_table = None
+        # self._execution_alias_table = None
+        # self._dependency_table = None
+        self._dataset_native = None
+        self._from_execution = None
+        self._all_dataset_properties = None
+        self._get_properties()
+
+    def _get_properties(self):
+        '''
+        Cache column names which may be queried.
+        '''
+        if self._dataset_native is None:
+            self._dataset_native = dict()
+            for c in self._tables['dataset'].c:
+                self._dataset_native['dataset.' + c.name] = is_orderable_type(c.type)
+
+        if  self._from_execution is None:
+            self._from_execution = dict()
+            for c in self._tables['execution'].c:
+                self._from_execution['execution.' + c.name] = is_orderable_type(c.type)
+
+    def list_dataset_properties(self):
+        '''
+        Returns a set of property names of the form <table_name>.<column_name>
+        <table_name> is either 'dataset' or another table name (e.g. 'execution')
+        for a table which may be joined with dataset.
+        '''
+        if self._all_dataset_properties:
+            return self._all_dataset_properties.keys()
+
+        all_properties = dict(self._dataset_native)
+        all_properties.update(self._from_execution)
+        self._all_dataset_properties = dict(all_properties)
+        return list(all_properties.keys())
+
+    def is_orderable_property(self, property_name):
+        if not self._all_dataset_properties:
+            list_dataset_properties()
+        if not property_name in self._all_dataset_properties.keys():
+            raise ValueException(f'Unknown property {property_name}')
+        return self._all_dataset_properties[property_name]
+
+    # Do we need this routine?  Will users want to query executions?
+    def list_execution_properties(self):
+        '''
+        Should return a list of property names, not necessarily identical
+        to column names in the execution table.
+        '''
+        pass
+
+    def _check_property_name(self, c):
+        '''
+        Given string representation of form <column_name> or
+        <table_name>.<column_name> check that it identifies a column belonging to the
+        dataset table or a table it references or is associated with
+        If input is just <column_name> and that name appears in more than one relevant
+        table, raise ValueException
+
+        Parameters
+        c      string   identifies possible column
+        Returns
+        table name, column name, column reference
+        '''
+        parts = c.split('.')
+        if len(parts) > 2:
+            raise ValueException(f'check_filter: "{f[0]}" is not a dataset property')
```

Comment:
error should be different from the one below so we can distinguish them.

can just be `raise ValueException(f'check_filter: "{c}" is not a value <table_name>.<column_name> combination')`

`f` should be `c` right?